version: '3.9'
services:
  social-recovery-service:
    # image: docker.swaywallet.io/social-recovery-service:latest
    build: 
      context: .
      args:
        appPort: ${SOCIAL_RECOVERY_SERVICE_PORT}
        nodeVersion: ${NODE_VERSION}
    ports:
      - ${SOCIAL_RECOVERY_SERVICE_PORT}:${SOCIAL_RECOVERY_SERVICE_PORT}
    env_file:
      - .env
    depends_on:
      db:
        condition: service_healthy
  db:
    image: postgres:16.1-bookworm
    restart: always
    environment:
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_DB=${DB_NAME}
    ports:
      - '5432:5432'
    volumes:
      - db:/var/lib/postgresql/data
    healthcheck:
      test:
        ["CMD", "pg_isready", "-U", "postgres", "-d", "social-recovery-service"]
      interval: 5s
      timeout: 10s
      retries: 5
  pgadmin:
    image: dpage/pgadmin4
    container_name: pgadmin4_container
    restart: always
    ports:
      - '8888:80'
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD}
    volumes:
      - pgadmin-data:/var/lib/pgadmin
    depends_on:
      db:
        condition: service_healthy
volumes:
  db:
    driver: local
  pgadmin-data:
    driver: local
